### PR TITLE
Integrates hydra-head updates in order to ensure that ACL resources are not persisted when Works/Collections are only built

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Deprecates  https://github.com/samvera/hyrax/compare/hydra-head-469; Please see https://github.com/samvera/hydra-head/pull/469 for reference

### CODE REVIEW

**Comment by jrgriffiniii** [comment]
Should this be blocked by a patch release for https://github.com/samvera/hydra-head?

**Comment by jrgriffiniii** [comment]
This is blocked by https://github.com/samvera/hydra-head/pull/474 (CI passes on Travis, the latest build is for a rebase).  One this has been merged and 10.6.1 released, this PR can be updated.

**Comment by no-reply** [comment]
is this still WIP? @jrgriffiniii

File: spec/services/hyrax/collections/migration_service_spec.rb [hunk 1]
```diff
@@ -14,11 +14,11 @@
 
   describe ".migrate_all_collections" do
     context 'when legacy collections are found (e.g. collections created before Hyrax 2.1.0)' do
-      let!(:col_none) { build(:typeless_collection, id: 'col_none', user: user, edit_users: [user.user_key], do_save: true) }
-      let!(:col_vu) { build(:typeless_collection, id: 'col_vu', user: user, edit_users: [user.user_key], read_users: [reader1.user_key, reader2.user_key], do_save: true) }
-      let!(:col_vg) { build(:typeless_collection, id: 'col_vg', user: user, edit_users: [user.user_key], read_groups: ['read_group_1', 'read_group_2'], do_save: true) }
-      let!(:col_mu) { build(:typeless_collection, id: 'col_mu', user: user, edit_users: [user.user_key, editor1.user_key, editor2.user_key], do_save: true) }
-      let!(:col_mg) { build(:typeless_collection, id: 'col_mg', user: user, edit_users: [user.user_key], edit_groups: ['edit_group_1', 'edit_group_2'], do_save: true) }
+      let!(:col_none) { build(:typeless_collection, user: user, edit_users: [user.user_key], do_save: true) }
```
Comment (cjcolvar) [inline spec/services/hyrax/collections/migration_service_spec.rb]: Follow on cleanup PR could switch these to `typeless_collection_lw`.

**Pull request merged**